src/DependencyInjection/CompilerPass/GenerateAdminServicesCompilerPass.php

```diff
@@ -61,10 +63,17 @@ public function process(ContainerBuilder $container)
                 $tags = $adminService->getTags();
                 $tags['sonata.admin'][0]['show_in_dashboard'] = 0;
                 $tags['sonata.admin'][0]['label'] = $id;
-                $adminService->setTags($tags);
 
-                $adminService->replaceArgument(0, $id);
-                $adminService->replaceArgument(1, $entityClassName);
+                if (count($adminService->getArguments()) > (is_a($adminService->getClass(), PageAdmin::class, true) ? 1 : 0)) {
```

Comment:
Ik vind deze conditie heel moeilijk leesbaar, idem hieronder. Is dit te versimpelen?

Reply:
```php
$maxArgumentsCount = is_a($adminService->getClass(), PageAdmin::class, true) ? 1 : 0;
if (count($adminService->getArguments()) > $maxArgumentsCount) {
```
Zoiets?

Reply:
Mja zoiets bijvoorbeeld. En wellicht hier een regeltje comment er bij waarom we dit checken. Dat het iig iets duidelijker wordt waarom we dit doen.